src/OpenConext/Profile/Entity/AuthenticatedUser.php

```diff
@@ -43,13 +45,47 @@
     /**
      * @param AssertionAdapter $assertionAdapter
      * @param EntityId[] $authenticatingAuthorities
+     *
      * @return AuthenticatedUser
+     * @throws RuntimeException
      */
     public static function createFrom(AssertionAdapter $assertionAdapter, array $authenticatingAuthorities)
     {
+        $attributes = [];
+
+        /** @var Attribute $attribute */
+        foreach ($assertionAdapter->getAttributeSet() as $attribute) {
+            $definition = $attribute->getAttributeDefinition();
+
+            // We need to replace the eduPersonTargetedID attribute value as that is a nested NameID attribute.
+            if ($definition->getName() === 'eduPersonTargetedID') {
+
+                /** @var \DOMNodeList[] $eptiValues */
+                $eptiValues = $attribute->getValue();
+                $eptiDomNodeList = $eptiValues[0];
+
+                if (!$eptiDomNodeList instanceof \DOMNodeList || $eptiDomNodeList->length !== 1) {
+                    throw new RuntimeException(
+                        sprintf(
+                            'EPTI attribute must contain exactly one NameID element as value, received: %s',
+                            print_r($eptiValues, true)
+                        )
+                    );
+                }
+
+                $eptiValue  = $eptiDomNodeList->item(0);
+                $eptiNameId = \SAML2_Utils::parseNameId($eptiValue);
+
+                $attributes[] = new Attribute($definition, [$eptiNameId['Value']]);
+                continue;
+            }
+
+            $attributes[] = $attribute;
```

Comment:
I'd suggest inverting the if statement, so that, when reading this code, you can quickly take the easy path `$attributes[]=$attribute;` out this for loop.